### PR TITLE
Add TypeScript

### PR DESCRIPTION
**Pull request by Meierschlumpf**
I've added typescript to the project and defined the types of the different variables.

### CODE REVIEW

**Comment by Meierschlumpf** [comment]
Hey @rtivital 
I've added the build step and polished the clustered tsconfig.json. About the exports field in package.json i'm not certain how to do. Could you help me there?

**Comment by rtivital** [comment]
The `main` field in the package.json must be the output of tsc compilation. That's pretty much it

File: package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "postcss-preset-mantine",
   "version": "1.3.0",
   "description": "PostCSS preset for Mantine (7.0+) applications",
-  "main": "preset.js",
+  "main": "preset.ts",
```
Comment (Meierschlumpf) [inline package.json:5]:
```suggestion
  "main": "dist/preset.js",
```
So I just need to change it like that? @rtivital

Comment (rtivital) [reply package.json:5]: yes, also make sure that dist folder is in gitignore

**Comment by rtivital** [comment]
Thanks!

**Pull request merged**